app/javascript/src/utils/compiler/translations.ts

```diff
@@ -8,9 +8,50 @@ export function convertTranslations(joinedItems: string, singleLanguageOverride:
   if (!get(selectedLanguages)?.length) return joinedItems
   if (!Object.keys(get(translationKeys) || {})?.length) return joinedItems
 
-  // Find @translate in content and replace with given translation key
+  joinedItems = replaceStaticTranslationKeys(joinedItems, singleLanguageOverride)
+  joinedItems = replaceDynamicTranslationKeys(joinedItems, singleLanguageOverride)
+
+  if (singleLanguageOverride) return joinedItems
+
+  // Array with custom string for Practice Range in each selected language
+  const customStringArrayForEachLanguage = get(selectedLanguages).map(language => `Custom String("${languageOptions[language].detect}")`)
+  const rule = `
+  rule("Workshop.codes Editor Dynamic Language - Set Languages") {
+    event { Ongoing - Global; }
+    actions { Global.WCDynamicLanguages = Array(${customStringArrayForEachLanguage.join(", ")}); }
+  }`
+
+  return joinedItems + rule
+}
+
+/**
+ * Replace @translate.static("Key") with "Value". Always compiles to a single language.
+ */
+function replaceStaticTranslationKeys(joinedItems: string, singleLanguageOverride: Language | null): string {
+  const staticRegex = /@translate\.static\(/g
+
   let match
+  while ((match = staticRegex.exec(joinedItems)) != null) {
+    const closing = getClosingBracket(joinedItems, "(", ")", match.index + 1)
+    if (closing < 0) continue
+
+    const full = joinedItems.slice(match.index, closing + 1)
+    const key = full.slice(full.indexOf("(") + 1, full.lastIndexOf(")")).replaceAll("\"", "")
+    const translation = getValueForLanguage(key, singleLanguageOverride || get(selectedLanguages)[0])
+
+    joinedItems = replaceBetween(joinedItems, `"${translation}"`, match.index, match.index + full.length)
+  }
+
+  return joinedItems
+}
+
+/**
+ * Replace @translate("Key") with Custom String("Value"), either compiled to a single language or a with dynamic languages.
+ */
+function replaceDynamicTranslationKeys(joinedItems: string, singleLanguageOverride: Language | null): string {
   const regex = /@translate/g
```

Comment:
Just making sure :eyes:

```suggestion
  const regex = /@translate(?!\.static)/g
```